### ISSUE TITLE
Mob not dropping items

### ISSUE THREAD

**Issue by ColdFlame53** [Slexom/earth2java#48]
The mob “Fuzzy sheep” only seems to drop xp. (1.17 Fabric)

**Comment by SaltSouls** [comment]
It isn't just the Fuzzy Sheep. I have killed multiple varieties of the base mobs(their earth counterparts) and have gotten no food drops or anything besides xp.

**Comment by Rayxanber** [comment]
Same issue here. It affects many of the mobs-pigs, cows, sheep.

**Comment by Footage2-Amply-Pounce** [comment]
I used to have the same issue on 1.17, but after updating to 1.17.1 and using different arguments to start my server all E2J mobs now drop loot properly. I'm not sure what fixed it, just updating to 1.17.1 or using the starting arguments, but you can see what I'm using and see if that helps you.
<details> <summary>Sever starting arguments</summary>
screen java -Xms16G -Xmx16G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -XX:G1NewSizePercent=40 -XX:G1MaxNewSizePercent=50 -XX:G1HeapRegionSize=16M -XX:G1ReservePercent=15 -XX:InitiatingHeapOccupancyPercent=20 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -jar server.jar nogui
</details>

Generated by [this tool](https://blog.airplane.gg/aikar-flags/) using [this guide](https://mcflags.emc.gs/).

**Comment by SaltSouls** [comment]
> I used to have the same issue on 1.17, but after updating to 1.17.1 and using different arguments to start my server all E2J mobs now drop loot properly. I'm not sure what fixed it, just updating to 1.17.1 or using the starting arguments, but you can see what I'm using and see if that helps you.
Okay, so JVM arguments have no effect on mob drops, I was playing on single player when this happened so arguments for a server wouldn't be causing the issue even if they could. If they could then something really dumb is going on with mob drops as those should be handled by loot tables and shouldn't rely on the user having certain arguments(So still a bug but now bigger).
A minor version change also shouldn't change this problem, though it might, but again mob drops shouldn't be effect as they are currently as again the game should be handling them via loot tables which should work across all minor/major versions as long as the code base ins't broken.

**Comment by Footage2-Amply-Pounce** [comment]
Absolutely, it is still a bug and, indeed, you are correct in that it's not the arguments that helped, I went into a SP world and still had proper E2J loot. However, despite the overall update being insignificant, something like an updated Fabric API could've fixed the issue for me.

<details>
  <summary>Here's my mod list, maybe that will help to narrow it down</summary>
[1.17]+Magic+Mirror+1.3.1
Adorn-2.1.1+1.17.1-fabric
advancements-enlarger-0.2.4
amecs-1.3.4+mc.1.17
appleskin-fabric-mc1.17-2.0.0
architects-palette-fabric-port-1.2
architectury-2.2.21-fabric
Artificial_Foliage_1.6_(FABRIC)_for_1.17-1.17.1
artisanat-2.1
autoconfig1u-3.3.1
axolotl-item-fix-1.1.1
backslot-1.1.13
backslotaddon-1.0.4
bclib-0.2.2
bedrockify-1.1.1+mc1.17
beehive-tooltips-1.3.0
beenfo-1.17.1-fabric0.36.1-1.3.2
betterbeds-1.2.0
BetterControls-1.17+-v1.2.0
better-end-0.10.4-pre
BetterMineshafts-Fabric-1.17-1.0.0
BetterStrongholds-Fabric-1.17-1.0
better-wandering-trader-1.7.0
bigbeacons-1.2.3
blockus-2.2.6+1.17.1
boats-and-beeps-1.0.1
branders-sulfurpotassiummod-1.17-1.1.1
BuildingWands-Fabric_1.17.1-1.5_beta2
Cardinal-Components-API-3.0.1
carrier-1.8.0
cavebiomes-0.6.3
charm-fabric-1.17-3.3.0
charmonium-fabric-1.17-3.3.0
clayrecipefabric-2.0.0
cleardespawn-1.17.1-fabric0.36.1-1.1.1
clientsidenoteblocks-2.0+1.17
cloth-api-2.0.54
cloth-config-5.0.34-fabric
Clumps-fabric-1.17-7.0.0.2
comforts-fabric-0.0.4-1.17
consistency_plus-1.17-0.4.0-RC1
CraftPresence-Fabric-1.17.1-Release-1.8.2
Croptopia-1.17-FABRIC-1.5.0
custom-portals-2.1.0
DarkPaintings-fabric-1.17-3.0.6
dawn-2.0.3
dazzle-2-1.17-1.0
death_controls-1.1.0
dragonloot-1.0.8
DualRiders-1.17-1.0.5
Dynocaps-1.8.1
Earth2Java-1.7.1+1.17
elementalcreepers-1.4.0
EnchantmentDescriptions-fabric-1.17-3.0.12
expandedstorage-7.0.0-alpha.33+1.17
fabric-api-0.37.0+1.17
fabric-enchantments-0.2.3
fabric-language-kotlin-1.6.2+kotlin.1.5.20
fabric-searchable-containers-1.17.1-1.3.0
fabrilous-updater-2.3
fallingleaves-1.7.2+1.17
FallingTree-1.17.1-2.12.1
fast-chest-1.2+1.17
ferritecore-3.0.1-fabric
flytre_lib-1.17-0.2.21
forgottenrecipes-fabric-1.0.1
grid-1.17.1-fabric0.36.1-1.6.5
helpwanted-1.1.0
horse-stats-vanilla-4.1.8
image2map-0.2.1
indrev-1.10.0-alpha.9
inmis-2.3.1-1.17
InventoryHUD-fabric-[1.17]-3.3.2
InventorySorter-1.7.9-1.17
invisiblights-fabric-7.1+1.17
itemscroller-fabric-1.17.1-0.15.0-dev.20210707.005506
Lakeside-1.1.0+1.17
lapisreserve-1.0.8
lifts-1.2-BETA+1.17
linkedstorage-1.3.11-1.17
malilib-fabric-1.17.1-0.10.0-dev.23
mcda-1.7.1
mcdar-1.3.1
mcdw-3.2.4
mcsa-2.3.2-1.17
megane-5.1.0
modmenu-2.0.2
morevillagers-FABRIC-1.17-2.0.2
netherite-items-1.17-1.1.0
notenoughcrashes-3.3.1+1.17-fabric
offhandoverhaul-1.1.1
onsoulfire-1.17-2
optifabric-1.11.18
pettable-1.2.0
PlayerPig-2.1
pling-1.4.0
PresenceFootsteps-r31-1.17-rc1
preview_OptiFine_1.17.1_HD_U_G9_pre27
replanter-fabric-1.17.1-1.3
repurposed_structures_fabric-2.1.0+1.17.1
respawnablepets-1.17-2
RoughlyEnoughItems-6.0.264-alpha-fabric
scorch-1.0.2
secretrooms-1.0.0+1.17
sihywtcamc-1.3.1+1.17.1
sihywtcamd-1.3.1+1.17.1
SimpleBackpack_Fabric-1.17.1-1.1.13
slotlock-1.0.4-BETA+1.17
soundphysics-1.1.1
speedyhopper-1.1-1.17-1
statuseffecttimer-1.0.2
stickyhopper-2.3-1.17-1
stitched-snow-1.0.3
tomsstorage-1.1.13-1.17.1
trident_revision-1.0.2
true-secret-rooms-1.1.4
TweakMyClient-fabric-mc1.17-1.1.5+build.18
twigs-1.0.1
vanillaplusbiomes-0.3.2
veinmining-fabric-0.0.10-1.17
villagernames-1.17-2.0.0
voicechat-fabric-1.17.1-1.0.6
Walkways-1.0.1
wanderingcollector-1.0.2+mc1.17
wirelessnetworks-1.1.2
witherite-0.0.7
wraith-silkspawners-1.0.2
wthit-fabric-3.7.0
Xaeros_Minimap_21.12.5.1_Fabric_1.17.1
XaerosWorldMap_1.14.5.2_Fabric_1.17.1
xlpackets-2.0-1.17
YungsApi-1.17-Fabric-11
</details>

**Comment by JuceDoesThings** [comment]
I also am having this issue. Many mobs aren't dropping food, which sucks in my hardcore world when I can only find broken Earth2Java mobs

**Comment by violetkitty87** [comment]
This is also happening on 1.21 fabric
